pandas/tests/indexes/test_multi.py

```diff
@@ -766,6 +766,14 @@ def test_from_arrays_empty(self):
                 ValueError, "Must pass non-zero number of levels/labels"):
             MultiIndex.from_arrays(arrays=[])
 
+        # 0 levels, names defined
+        result = MultiIndex.from_arrays(arrays=[],
```

Comment:
so ``.from_arrays`` will always factorize and hence the ``Float64Index`` result. This is a separate issue (and we need to do this for 0.21 as its technically an API change, pls open an issue). We can do the empty check on all levels  to avoid the factorization (as its proper to have these be ``object`` dtype and not float).